static/main.ts

```diff
@@ -22,42 +22,52 @@
-var analytics = require('./analytics').ga;
-
-require('whatwg-fetch');
-// eslint-disable-next-line requirejs/no-js-extension
-require('popper.js');
-require('bootstrap');
-
-var Sharing = require('./sharing').Sharing;
-var _ = require('underscore');
-var $ = require('jquery');
-var GoldenLayout = require('golden-layout');
-var Components = require('./components');
-var url = require('./url');
-var clipboard = require('clipboard');
-var Hub = require('./hub').Hub;
-var Sentry = require('@sentry/browser');
-var Settings = require('./settings').Settings;
-var local = require('./local');
-var Alert = require('./widgets/alert').Alert;
-var themer = require('./themes');
-var motd = require('./motd');
-var jsCookie = require('js-cookie');
-var SimpleCook = require('./widgets/simplecook').SimpleCook;
-var HistoryWidget = require('./widgets/history-widget').HistoryWidget;
-var History = require('./history');
-var Presentation = require('./presentation').Presentation;
-var setupSiteTemplateWidgetButton = require('./widgets/site-templates-widget').setupSiteTemplateWidgetButton;
-
-var logos = require.context('../views/resources/logos', false, /\.(png|svg)$/);
-
-var siteTemplateScreenshots = require.context('../views/resources/template_screenshots', false, /\.png$/);
-
-if (!window.PRODUCTION) {
+import {ga as analytics} from './analytics';
+
+import 'whatwg-fetch';
+import 'popper.js'; // eslint-disable-line requirejs/no-js-extension
+import 'bootstrap';
+
+import $ from 'jquery';
+import _ from 'underscore';
+
+import GoldenLayout from 'golden-layout';
+import JsCookie from 'js-cookie';
+import clipboard from 'clipboard';
+import * as Sentry from '@sentry/browser';
+
+// We re-assign this
+let jsCookie = JsCookie;
+
+import {Sharing} from './sharing';
+import * as Components from './components';
+import * as url from './url';
+import {Hub} from './hub';
+import {Settings, SiteSettings} from './settings';
+import * as local from './local';
+import {Alert} from './widgets/alert';
+import * as themer from './themes';
+import * as motd from './motd';
+import {SimpleCook} from './widgets/simplecook';
+import {HistoryWidget} from './widgets/history-widget';
+import * as History from './history';
+import {Presentation} from './presentation';
+import {setupSiteTemplateWidgetButton} from './widgets/site-templates-widget';
+import {options} from './options';
+import {unwrap} from './assert';
+
+import {Language, LanguageKey} from '../types/languages.interfaces';
+import {CompilerExplorerOptions} from './global';
+import {ComponentConfig, EmptyCompilerState, StateWithId, StateWithLanguage} from './components.interfaces';
+
+import * as utils from '../lib/common-utils';
+
+const logos = require.context('../views/resources/logos', false, /\.(png|svg)$/);
+
+const siteTemplateScreenshots = require.context('../views/resources/template_screenshots', false, /\.png$/);
+
+if (!window.PRODUCTION && !options.embedded) {
```

Comment:
Fixes #4521, cc @RubenRBS It looks like `global.sinon` isn't defined for embedded in static/tests/motd

Reply:
Ohh, good catch